src/app/modules/main/chat_section/controller.nim

```diff
@@ -146,6 +171,13 @@ proc init*(self: Controller) =
       self.contactService, self.chatService, self.communityService, self.messageService, self.gifService,
       self.mailserversService, setChatAsActive = true)
 
+  self.events.on(SIGNAL_SHARED_KEYCARD_MODULE_USER_AUTHENTICATED) do(e: Args):
+    let args = SharedKeycarModuleArgs(e)
+    if args.uniqueIdentifier != UNIQUE_MAIN_MODULE_AUTH_IDENTIFIER:
+      return
+    if self.tmpRequestToJoinCommunityId == self.sectionId:
+      self.delegate.onUserAuthenticated(args.pin, args.password, args.keyUid)
```

Comment:
> > I'm confused, it doesn't seem like you get the password yet. Are you gonna do that in another PR?

@jrainville this is where the password is emitted by the modal.
I know it's confusing and not very straight forward, but the idea is essentially that you call `authenticate()` which will do whatever is needed to open a modal and perform authentication and once the user has authenticated, this even here is processed in which case we get access to `args.password`.

Lots of implicitly in here, but that's how it is